core/src/main/scala/kafka/raft/KafkaMetadataLog.scala

```diff
@@ -677,4 +692,20 @@ object KafkaMetadataLog extends Logging {
       Snapshots.deleteIfExists(logDir, snapshotId)
     }
   }
+
+  trait SnapshotDeletionReason {
+    def logReason(snapshotId: OffsetAndEpoch): Unit
+  }
+
+  final case class RetentionMsBreach() extends SnapshotDeletionReason {
+    override def logReason(snapshotId: OffsetAndEpoch): Unit = {
+      info(s"Marking snapshot $snapshotId for deletion because the age is too old")
```

Comment:
Should this function returns a `String` instead log calling `info` directly?

The reason should include this information: `now - timestamp > config.retentionMillis`. From https://github.com/apache/kafka/pull/15450/files#diff-b332f85b04775c821226b6f704e91d51f9647f29ba73dace65b99cf36f6b9ceaR455

For Example,
`s"the snapshot's timestamp ($timestamp) is now ($now) older than the retention (${config.retentionMillis})"`

Reply:
Do you mind give me a hint on how to pass the time difference (i.e., now - timestamp) to the `RetentionMsBreanch` function? I tried to modify the `shouldClean` function to pass a tuple of (Boolean, Long) to pass the time difference as the second factor, but was blocked. Could you give me some hints to unblock? Thanks!